data-migrator/core/src/main/java/io/camunda/migration/data/converter/DecisionInstanceConverter.java

```diff
@@ -28,23 +28,25 @@ public class DecisionInstanceConverter {
   protected VariableService variableService;
 
   public DecisionInstanceDbModel apply(HistoricDecisionInstance decisionInstance,
+                                       String decisionInstanceId,
                                        Long decisionDefinitionKey,
                                        Long processDefinitionKey,
                                        Long decisionRequirementsDefinitionKey,
                                        Long processInstanceKey,
                                        Long rootDecisionDefinitionKey,
                                        Long flowNodeInstanceKey,
                                        String flowNodeId) {
-    Long decisionInstanceKey = getNextKey();
+
+    long decisionInstanceKey = Long.parseLong(decisionInstanceId.split("-")[0]);
+
     return new DecisionInstanceDbModel.Builder()
         .partitionId(C7_HISTORY_PARTITION_ID)
-        .decisionInstanceId(String.format("%d-%s", decisionInstanceKey, decisionInstance.getId()))
+        .decisionInstanceId(decisionInstanceId)
         .decisionInstanceKey(decisionInstanceKey)
-        .state(null) // TODO https://github.com/camunda/camunda-bpm-platform/issues/5370
+        .state(DecisionInstanceEntity.DecisionInstanceState.EVALUATED) // TODO https://github.com/camunda/camunda-bpm-platform/issues/5370
```

Comment:
❓ I think this TODO is resolved with the current changes. Also this TODO point to a completed issue.

Reply:
This is already resolved.